src/Confluent.Kafka/Timestamp.cs

```diff
@@ -36,16 +36,38 @@ public struct Timestamp
         /// <summary>
         ///     Initializes a new instance of the Timestamp structure.
         /// </summary>
+        /// <param name="unixTimestampMs">
+        ///     The unix millisecond timestamp.
+        /// </param>
+        /// <param name="type">
+        ///     The type of the timestamp.
+        /// </param>
+        public Timestamp(long unixTimestampMs, TimestampType type)
+        {
+            Type = type;
+            UnixTimestampMs = unixTimestampMs;
+        }
+
+        /// <summary>
+        ///     Initializes a new instance of the Timestamp structure.
+        ///     Prefer the (long, TimestampType) overload if possible.
```

Comment:
> Message (returned by consumer and delivery handler) contains TimeStamp, so I would keep the datetime property for easy usage

the released version contains a / 1000 bug (in two places) which means timestamp values are always wrong (fixed in master, but nothing released yet) - this means no one is relying on this yet, so I think we can get away with just changing it... however obsolete is fine too, I wouldn't bother changing it.

Perhaps we should have a constructor that takes `DateTimeOffset` as well? (feel free to add, but I'll still merge without).